### PR TITLE
odo Release Notes, Known Issues and typos fixes

### PR DESCRIPTION
**Pull request by yhontyk**
- branch/enterprise-4.2
- reviewed and approved by SMEs, QE (KIs were compiled and provided by the QE)
- fixed typos pointed out in https://github.com/openshift/openshift-docs/pull/17208

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@openshift/team-documentation PTAL.

File: _topic_map.yml [hunk 1]
```diff
@@ -1013,6 +1013,8 @@ Topics:
     File: configuring-the-odo-cli
   - Name: odo CLI reference
     File: odo-cli-reference
+  - Name: odo Release Notes
```
Comment (jboxman) [inline _topic_map.yml]: s/Release Notes/Release notes/

sentence case

File: modules/openshift-developer-cli-odo-notable-features.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-notable-features_{context}"]
+= Notable Features in {odo-title}{nbsp}{odo-ver}
```
Comment (jboxman) [inline modules/openshift-developer-cli-odo-notable-features.adoc]: s/Notable Features/Notable features/

sentence case

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[id='odo-release-notes']
+= {odo-title}{nbsp}{odo-ver} Release Notes
```
Comment (jboxman) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc]: @boczkowska, I haven't seen `{nbsp}` used before. Why doesn't a normal space work here?

Comment (bergerhoffer) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc]: @jboxman Some products use the {nbsp} to keep the product name and version together (instead of letting them split across lines). It doesn't look like we do in our guidelines, but I think this is fine here.

File: modules/openshift-developer-cli-odo-getting-support.adoc [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-getting-support_{context}"]
+= Getting Support
+
+.Documentation:
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-getting-support.adoc]: Shouldn't use a colon at the end when you're doing a title heading (.Documentation).

Same for the two headings below.

File: modules/openshift-developer-cli-odo-getting-support.adoc [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-getting-support_{context}"]
+= Getting Support
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-getting-support.adoc]: Getting support

File: modules/openshift-developer-cli-odo-getting-support.adoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+.Documentation:
+
+If you have found an error or have suggestions for improving the documentation, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose *{product-title}* product type and *Documentation* component type.
+
+.Product:
+
+If you have found an error, encountered a bug or have suggestions for improving the functionality of {odo-title}, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose *{product-title}* product type and *{odo-title}* component type.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-getting-support.adoc]: Choose -> Choose the

File: modules/openshift-developer-cli-odo-getting-support.adoc [hunk 7]
```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-getting-support_{context}"]
+= Getting Support
+
+.Documentation:
+
+If you have found an error or have suggestions for improving the documentation, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose *{product-title}* product type and *Documentation* component type.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-getting-support.adoc]: Choose -> Choos the

File: modules/openshift-developer-cli-odo-getting-support.adoc [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+.Product:
+
+If you have found an error, encountered a bug or have suggestions for improving the functionality of {odo-title}, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose *{product-title}* product type and *{odo-title}* component type.
+
+Provide as many details in the description of the issue as possible.
+
+.Getting in touch with the product team:
+
+Follow the news in {odo-title} link: https://groups.google.com/forum/#!forum/odo-users[Google Group].
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-getting-support.adoc]: Suggest either putting these two things in bullets, or combining them into a single sentence like:

```
Follow news in the {odo-title} link:https://groups.google.com/forum/#!forum/odo-users[Google Group] or subscribe to the `odo-users@googlegroups.com` mailing list.
```

(Also note that you need to remove the space between "link:" and "https://...")

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-known-issues_{context}"]
+= Known Issues
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760573[Bug 1760573] An active pointer is not pointing to a current active project.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760574[Bug 1760574] A deleted namespace is being listed in `odo project get` command.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760575[Bug 1760575] `odo app delete' removes application components but not services.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: "odo app delete" needs a backtick (`) instead of a single quote (') after it.

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 10]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-known-issues_{context}"]
+= Known Issues
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760573[Bug 1760573] An active pointer is not pointing to a current active project.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760574[Bug 1760574] A deleted namespace is being listed in `odo project get` command.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: listed in -> listed in the

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760577[Bug 1760577] `odo push` does not delete OpenShift objects when component name is changed.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760578[Bug 1760578] `odo watch` does not fail with error message for git based components but it should.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760580[Bug 1760580] `catalog list components` does not list out component of a specified namespace.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760583[Bug 1760583] `odo config unset` does not unset an environment variable but says it did.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760585[Bug 1760585] `odo delete --all` deletes the`$HOME/.odo` folder is deleted when run from `$HOME`.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: Missing space between "the" and "`$HOME/.odo`", so the backticks are rendering in the text.

And something is wrong with this sentence. Should it be...

```
deletes the`$HOME/.odo` folder when run from `$HOME`.
```

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760586[Bug 1760586] `odo delete` starts an infinite loop after project is delete and component name is set.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: after project is delete and component name -> after a project is deleted an a component name

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760587[Bug 1760587] {odo-title} reports an invalid flag when a non-existent command is used.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760588[Bug 1760588] `odo service create` crashes when run in Cygwin.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760589[Bug 1760589] Autocompletion does not work for  `--context` in `odo push`.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760590[Bug 1760590] In Git BASH for Windows `odo login -u developer` does not hide typed password when requested.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: typed password -> a typed password

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760573[Bug 1760573] An active pointer is not pointing to a current active project.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760574[Bug 1760574] A deleted namespace is being listed in `odo project get` command.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760575[Bug 1760575] `odo app delete' removes application components but not services.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760577[Bug 1760577] `odo push` does not delete OpenShift objects when component name is changed.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760578[Bug 1760578] `odo watch` does not fail with error message for git based components but it should.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: fail with error message for git based -> fail with an error message for git-based

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+[id="odo-known-issues_{context}"]
+= Known Issues
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760573[Bug 1760573] An active pointer is not pointing to a current active project.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760574[Bug 1760574] A deleted namespace is being listed in `odo project get` command.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760575[Bug 1760575] `odo app delete' removes application components but not services.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760577[Bug 1760577] `odo push` does not delete OpenShift objects when component name is changed.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: when component name -> when the component name

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760578[Bug 1760578] `odo watch` does not fail with error message for git based components but it should.
+
+* link:https://bugzilla.redhat.com/show_bug.cgi?id=1760580[Bug 1760580] `catalog list components` does not list out component of a specified namespace.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: Should this be "does not list out components of a specified namespace."?

File: modules/openshift-developer-cli-odo-notable-features.adoc [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+{odo-title}{nbsp}{odo-ver} offers a simple way to develop applications on {product-title}. {odo-title} is completely client-based and requires no server within {product-title} cluster.
+
+{odo-title}{nbsp}{odo-ver} comes with:
+
+* Simple syntax and design centered around concepts familiar to developers such as projects, applications and components.
+* Compatibility with any language or runtime within the OpenShift catalog of component types.
+* Capability to add custom component types - custom image builders.
+* Official support for Java and Node.js component types.
+* Service catalog which allows users to connect applications deployed on {product-title} cluster to the Template Service Broker.
+* `odo service create` - an interactive mode which guides users through the process of the creation of applications.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-notable-features.adoc]: Suggest "of the creation of applications" -> "of creating applications"

File: modules/openshift-developer-cli-odo-notable-features.adoc [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+* `odo watch` - a feature which enables automatic detection of changes to the local code and application of the changes to the component automatically in real-time.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-notable-features.adoc]: Suggest rewording to something like "a feature that automatically detects changes to local code and applies the changes to the component in real time."

File: modules/openshift-developer-cli-odo-notable-features.adoc [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+[id="odo-notable-features_{context}"]
+= Notable Features in {odo-title}{nbsp}{odo-ver}
+
+{odo-title}{nbsp}{odo-ver} offers a simple way to develop applications on {product-title}. {odo-title} is completely client-based and requires no server within {product-title} cluster.
+
+{odo-title}{nbsp}{odo-ver} comes with:
+
+* Simple syntax and design centered around concepts familiar to developers such as projects, applications and components.
+* Compatibility with any language or runtime within the OpenShift catalog of component types.
+* Capability to add custom component types - custom image builders.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-notable-features.adoc]: Suggest maybe "using" instead of the dash here? 

`Capability to add custom component types using custom image builders.`

File: modules/openshift-developer-cli-odo-notable-features.adoc [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+* Official support for Java and Node.js component types.
+* Service catalog which allows users to connect applications deployed on {product-title} cluster to the Template Service Broker.
+* `odo service create` - an interactive mode which guides users through the process of the creation of applications.
+* `odo watch` - a feature which enables automatic detection of changes to the local code and application of the changes to the component automatically in real-time.
+
+See the {odo-title} CLI reference documentation to learn the details.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-notable-features.adoc]: Is this supposed to be for all of the features, or just the odo watch item? If it's the odo watch item, needs a + on the line above to keep it with that bullet. Otherwise I think you could just remove this as it doesn't seem to provide much detail for features.

File: modules/openshift-developer-cli-odo-technology-preview.adoc [hunk 21]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-technology-preview_{context}"]
+= List of features in Technology Preview in {odo-title}{nbsp}{odo-ver}:
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-technology-preview.adoc]: Suggest rewording and removing colon, to something like:

```
Technology Preview features in {odo-title}{nbsp}{odo-ver}
```

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 3 above)
Comment (bergerhoffer) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc]: And sentence case here

**Review by bergerhoffer — COMMENTED** [review]
@boczkowska Just a few little things, but overall looks good!

File: modules/openshift-developer-cli-odo-getting-support.adoc (hunk 8 above)
Comment (vikram-redhat) [reply modules/openshift-developer-cli-odo-getting-support.adoc]: @boczkowska @bergerhoffer 

I am uncomfortable about this link here. Is the discussion in this group RH supported? If not, we should remove it, or conditionalize this for OKD (upstream) only.

@sferich888 if you have any thoughts here.

Comment (boczkowska) [reply modules/openshift-developer-cli-odo-getting-support.adoc]: @vikram-redhat You're right, it's intended for upstream and is not RH supported. I'm removing it.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 3 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc]: @jboxman what @bergerhoffer said! I can remove it but AFAIK it renders fine so shouldn't be a problem.

File: modules/openshift-developer-cli-odo-known-issues.adoc [hunk 22]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/odo-release-notes.adoc
+
+[id="odo-known-issues_{context}"]
+= Known Issues
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-odo-known-issues.adoc]: Known issues

**Comment by bergerhoffer** [comment]
LGTM!
